Revert changes introduced in Magento 2.4.5 which causing issues with "Stock Status" for grouped product because it's loading data from legacy stock.
Original commit: https://github.com/magento/inventory/commit/0814fe7edfb9f4344494d941b479509c3f555e85
Similar related issue: https://github.com/magento/inventory/issues/3350

diff --git a/Indexer/SelectBuilder.php b/Indexer/SelectBuilder.php
index 019aadc..25b33d8 100644
--- a/Indexer/SelectBuilder.php
+++ b/Indexer/SelectBuilder.php
@@ -120,17 +120,23 @@ class SelectBuilder
             'child_stock.sku = child_product_entity.sku',
             [
                 IndexStructure::QUANTITY => 'SUM(child_stock.quantity)',
-                IndexStructure::IS_SALABLE => 'IF(inventory_stock_item.is_in_stock = 0, 0,
-                MAX(child_stock.is_salable))',
+                // BEGIN PATCH
+                // Revert changes introduced in Magento 2.4.5 which causing issues with "Stock Status" for grouped product because it's loading data from legacy stock.
+                // IndexStructure::IS_SALABLE => 'IF(inventory_stock_item.is_in_stock = 0, 0,
+                // MAX(child_stock.is_salable))',
+                IndexStructure::IS_SALABLE => 'MAX(child_stock.is_salable)',
+                // END PATCH
             ]
         )->joinInner(
             ['child_filter_product_entity' => $this->resourceConnection->getTableName('catalog_product_entity')],
             "child_filter_product_entity.entity_id = parent_link.linked_product_id",
             []
-        )->joinLeft(
-            ['inventory_stock_item' => $this->resourceConnection->getTableName('cataloginventory_stock_item')],
-            'inventory_stock_item.product_id = parent_product_entity.entity_id',
-            []
+        // BEGIN PATCH
+        // )->joinLeft(
+        //     ['inventory_stock_item' => $this->resourceConnection->getTableName('cataloginventory_stock_item')],
+        //     'inventory_stock_item.product_id = parent_product_entity.entity_id',
+        //     []
+        // END PATCH
         )->where(
             'parent_link.link_type_id = ' . Link::LINK_TYPE_GROUPED
         )->group(
